R/backend-redshift.R

```diff
@@ -76,6 +81,23 @@ sql_translation.RedshiftConnection <- function(con) {
           win_current_frame()
         )
       },
+      # https://docs.aws.amazon.com/redshift/latest/dg/r_LISTAGG.html
+      str_flatten = function(x, collapse) {
+        order <- win_current_order()
+        if(length(order) > 0){
+          sql <- build_sql(sql_expr(LISTAGG(!!x, !!collapse)),
+                           " WITHIN GROUP (ORDER BY ", order, ")")
```

Comment:
Are there other functions in redshift that use this syntax? I'm surprised that (say) windowed `mean()` doesn't need the same syntax? Or is `LISTAGG()` the same sort of function as `PERCENTILE_DISC()`?

Reply:
From searching the documentation only the `LISTAGG()`, `PERCENTILE_DISC()`, `PERCENTILE_CONT()`, and `ST_COLLECT()`  functions in redshift require the `WITHIN GROUP` clause for the ordering.  The remaining windowed functions use the more general `OVER (PARTITION BY ... ORDER BY ...)` clause. 

Unfortunately, I cannot explain why `LISTAGG()` in particular uses the `WITHIN GROUP` syntax. But this syntax seems to be shared by other database types (e.g.  [Oracle](https://docs.oracle.com/cd/E11882_01/server.112/e41084/functions089.htm#SQLRF30030)).

Reply:
Thanks for the investigation! Unfortunately there seem to be few satisfying explanation for why SQL things are the way they are.